casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -224,15 +227,37 @@ class RuntimeManager private (val emptyStateHash: ByteString, runtimeContainer:
 object RuntimeManager {
   type StateHash = ByteString
 
-  def fromRuntime(active: Runtime): RuntimeManager = {
+  def fromRuntime(active: Runtime)(implicit scheduler: Scheduler): RuntimeManager = {
     active.space.clear()
     active.replaySpace.clear()
-    val hash       = ByteString.copyFrom(active.space.createCheckpoint().root.bytes.toArray)
-    val replayHash = ByteString.copyFrom(active.replaySpace.createCheckpoint().root.bytes.toArray)
-    assert(hash == replayHash)
+
+    //populate registry root with empty map
+    val emptyMap     = Par().withExprs(List(Expr(EMapBody(ParMap(Nil)))))
+    val registryRoot = Par().withIds(List(Registry.registryRoot))
+    val produceTerm  = Par().withSends(List(Send(registryRoot, List(emptyMap))))
+    implicit val rand = Blake2b512Random(
+      Array[Byte](3, 1, 4, 1, 5, 9, 2, 6, 5, 3, 5, 8, 9, 7, 9, 3)
+    )
+    val max       = Cost(Long.MaxValue)
+    val zero      = Cost(0L)
+    val emptyChPt = active.space.createCheckpoint()
+    val registryInit: Task[Array[Byte]] = for {
+      _          <- active.reducer.setAvailablePhlos(max)
+      _          <- active.reducer.inj(produceTerm)
+      checkpoint = active.space.createCheckpoint()
+      hash       = checkpoint.root.bytes.toArray
+      _          <- active.reducer.setAvailablePhlos(zero)
+      _          <- active.replayReducer.setAvailablePhlos(max)
+      _          = active.replaySpace.rig(emptyChPt.root, checkpoint.log)
+      _          <- active.replayReducer.inj(produceTerm)
```

Comment:
Calling `rig` only tells the replay rspace what it should expect. Nothing actually happens until to give it the terms though. This is where the "unused comm event" error comes from, if you create a checkpoint (indicating that you are done replaying) but the replay still expected there to be more terms then it will throw that error. `rig` also sets things up so that comm events happen in the same way when the terms are actually added.

Reply:
Thank you very much sir.